fastlane/lib/fastlane/actions/run_tests.rb

```diff
@@ -14,7 +14,6 @@ def self.run(values)
 
         begin
           destination = values[:destination] # save destination value which can be later overridden
-          Scan.config = values # we set this here to auto-detect missing values, which we need later on
```

Comment:
Absolutely sure this is not needed for anything? We usually don't just have stuff like that in the codebase if it doesn't have a purpose...

Reply:
Scan.config is again set with same `values` in the next line when it is passed to https://github.com/fastlane/fastlane/blob/4e3417ddeeee5958e83c53c65ed561cff306a972/fastlane/lib/fastlane/actions/run_tests.rb#L23  as options which is again setting scan.config.. https://github.com/fastlane/fastlane/blob/4e3417ddeeee5958e83c53c65ed561cff306a972/scan/lib/scan/manager.rb#L8

Reply:
if there is any usecase i would definitely like to test it with this change.. I have tested locally with both fastfile and scanfile .

Reply:
This `Scan.config = values` is actually needed here. Setting `Scan.config` here calls `DetectValues.set_additional_default_values` which is needed for the `test_summary_filenames` method call below.

If anything, the second `Scan.config=` in `Scan::Manager.new.work` shouldn't be called but we can't delete that one because that one is needed when calling `scan` from the CLI.

So this `Scan.config` that is called first in this action is needed for a little bit of extra stuff that the action does that `Scan::Manager.new.work` doesn't do.

Sooooooo... if anything, this extra behavior should maybe get moved into `Scan::Manager.new.work` so that the CLI version gets this extra output and we can remove the second call 🙃

Reply:
that sounds interesting and should the trick .. will test that locally.. thanks for the detail explanation .. much appreciated

Reply:
@joshdholtz any suggestions on how i can return `plist_files_before ` from `manager` to `run_tests` as manager is already returning 
https://github.com/fastlane/fastlane/blob/4e3417ddeeee5958e83c53c65ed561cff306a972/scan/lib/scan/manager.rb#L18

i was thinking of adding `plist_files_before` to Scan.cache by creating new method in  DetectValues.set_additional_default_values ?